### PR TITLE
Adding a script that will run e2e tests on our jenkins builder.

### PR DESCRIPTION
**Pull request by vishh**
The script should make updates to the test infra explicit.

### CODE REVIEW

File: build/jenkins_e2e.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# e2e-cadvisor-coreos-beta
+# - docker 1.9.1
+# e2e-cadvisor-rhel-7
+# - red hat 7
+# - docker 1.9.1
+
+./runner --logtostderr --test-retry-count=8 --test-retry-whitelist=integration/runner/retrywhitelist.txt \
+e2e-cadvisor-ubuntu-trusty \
+e2e-cadvisor-coreos-beta \
+e2e-cadvisor-container-vm-v20151215 \
```
Comment (timstclair) [inline build/jenkins_e2e.sh]: No hostnotes for this one?

File: build/jenkins_e2e.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+e2e-cadvisor-container-vm-v20160127 \
+e2e-cadvisor-rhel-7
+
+# TODO: enable when docker 1.10  is working
+# e2e-cadvisor-ubuntu-trusty-docker110
```
Comment (timstclair) [inline build/jenkins_e2e.sh]: is this the same as `e2e-cadvisor-ubuntu-trusty-docker1-10`? Which is correct?

File: build/jenkins_e2e.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# e2e-cadvisor-rhel-7
+# - red hat 7
+# - docker 1.9.1
+
+./runner --logtostderr --test-retry-count=8 --test-retry-whitelist=integration/runner/retrywhitelist.txt \
+e2e-cadvisor-ubuntu-trusty \
+e2e-cadvisor-coreos-beta \
+e2e-cadvisor-container-vm-v20151215 \
+e2e-cadvisor-container-vm-v20160127 \
+e2e-cadvisor-rhel-7
```
Comment (timstclair) [inline build/jenkins_e2e.sh]: I think it would be more readable if you make this list a variable

File: build/integration.sh [hunk 4]
```diff
@@ -14,6 +14,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if [[ ! -z "${JENKINS_HOME-}" ]]
```
Comment (timstclair) [inline build/integration.sh]: I think `! -z` is equivalent to `-n`?

File: build/integration.sh [hunk 5]
```diff
@@ -14,6 +14,10 @@
...
+then 
+    exec ./jenkins_e2e.sh
```
Comment (timstclair) [inline build/integration.sh]: I think you need to call exit after this

File: build/integration.sh (hunk 4 above)
Comment (jimmidyson) [inline build/integration.sh]: Is that extra `-` required?

File: build/integration.sh (hunk 5 above)
Comment (jimmidyson) [inline build/integration.sh]: Don't think so - exec replaces current shell process & doesn't return to caller iirc.

Not sure why you need to exec it though? Just call it normally? Or are you relying on the exit code of the exec script?

**Comment by jimmidyson** [comment]
Personally if prefer to move to a matrix build for testing on multiple distros. That way is easier to identify problems & build output is more readable. What do you think?

**Comment by vishh** [comment]
@jimmidyson: Matrix build is what we need. Any ideas on how to get there?

File: build/integration.sh (hunk 5 above)
Comment (vishh) [inline build/integration.sh]: The exit code is the reason for exec. I can catch the exit code and return that too, if exec has issues.

Comment (jimmidyson) [inline build/integration.sh]: I have no problem with exec here.

**Comment by jimmidyson** [comment]
For matrix build you would need to set up each node as a Jenkins slave & then set up the matrix to run on each slave.

**Comment by vishh** [comment]
We will have to periodically upgrade docker version and the base distributions. @pwittrock and I were brainstorming today to crowdsource testing if possible, similar to Go's [build dashboard](https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=go%20build%20dashboard). WDYT?

I will address the comments in this PR now. This PR will help with the process until we have a better test infra. Disagree?

File: build/jenkins_e2e.sh (hunk 1 above)
Comment (vishh) [inline build/jenkins_e2e.sh]: Fixed.

File: build/jenkins_e2e.sh (hunk 2 above)
Comment (vishh) [inline build/jenkins_e2e.sh]: `110` is the valid name.

File: build/jenkins_e2e.sh (hunk 3 above)
Comment (vishh) [inline build/jenkins_e2e.sh]: Done

File: build/integration.sh (hunk 4 above)
Comment (vishh) [inline build/integration.sh]: Valid comments. Updated. PTAL.

**Comment by vishh** [comment]
@timstclair @timstclair: PTAL.

**Comment by vishh** [comment]
I want to merge this PR before adding a docker v1.10 node.

**Comment by jimmidyson** [comment]
LGTM

**Pull request merged**